src/world/Action/Action.cpp

```diff
@@ -583,13 +590,72 @@ void Action::Action::buildActionResults()
       shouldRestoreMP = false;
     }
   }
+
+  // If we hit an enemy
+  if( m_hitActors.size() > 0 && getHitChara()->getObjKind() != m_pSource->getObjKind() )
+  {
+    m_pSource->removeStatusEffectByFlag( Common::StatusEffectFlag::RemoveOnSuccessfulHit );
+  }
+
+  handleJobAction();
+
+  if( m_lutEntry.statuses.caster.size() > 0 || m_lutEntry.statuses.target.size() > 0 )
+    handleStatusEffects();
+
   m_actionResultBuilder->sendActionResults( m_hitActors );
 
   // TODO: disabled, reset kills our queued actions
   // at this point we're done with it and no longer need it
   // m_effectBuilder.reset();
 }
 
+void Action::Action::handleStatusEffects()
+{
+  auto pActionBuilder = getActionResultBuilder();
+
+  if( !pActionBuilder )
+    return;
+
+  if( isComboAction() && !isCorrectCombo() )
+    return;
+
+  // handle caster statuses
+  if( m_lutEntry.statuses.caster.size() > 0 )
+  {
+    for( auto& status : m_lutEntry.statuses.caster )
+    {
+      pActionBuilder->applyStatusEffectSelf( status.id, status.duration, 0, status.modifiers, status.flag, true );
+    }
+  }
+
+  // handle hit actor statuses
+  if( m_lutEntry.statuses.target.size() > 0 && m_hitActors.size() > 0 )
+  {
+    for( auto& actor : m_hitActors )
+    {
+      for( auto& status : m_lutEntry.statuses.target )
+      {
+        pActionBuilder->applyStatusEffect( actor, status.id, status.duration, 0, status.modifiers, status.flag, true );
+      }
+
+      if( actor->getStatusEffectMap().size() > 0 )
+        actor->onActionHostile( m_pSource );
+    }
+  }
+}
+
+void Action::Action::handleJobAction()
+{
+  switch( m_pSource->getClass() )
+  {
+    case ClassJob::Warrior:
+    {
+      Warrior::onAction( *m_pSource->getAsPlayer(), *this );
```

Comment:
i'm okay with this as is for now, but i think it would probably better to have an array or otherwise that's just indexed by classjob (which also lets you pick up the base classjob impl easy) instead of this - i get this is wip-ish though so i'm not too worried about it at the moment